src/main/java/frc/robot/subsystems/Drivetrain.java

```diff
@@ -78,10 +82,58 @@ public void configure() {
     super.configure();
   }
 
+  /**
+   * Calculate which pose from an array has the closest rotation to the robot's
+   * current pose. If multiple poses have the same rotation, the last one in the
+   * list will be returned.
+   * 
+   * @param poses The list of poses to check
+   * @return The last pose in the list with the closest rotation
+   */
+  public Pose2d getClosestPoseByRotation(List<Pose2d> poses) {
+    Pose2d closestPose = poses.get(0);
+    double smallestDifference = Math.abs(getRotation().minus(closestPose.getRotation()).getRadians());
+    for (Pose2d pose : poses) {
+      double difference = Math.abs(getRotation().minus(pose.getRotation()).getRadians());
+      if (difference < smallestDifference) {
+        smallestDifference = difference;
+        closestPose = pose;
+      }
+    }
+    return closestPose;
+  }
+
+  public Pose2d getRobotPose() {
+    return getPose();
+  }
+
+  public Pose2d getFrontPose() {
+    return getPose().plus(new Transform2d(Units.Inches.of(0), Units.Inches.of(14.5), new Rotation2d(0)));
+  }
+
+  public Pose2d getBackPose() {
+    return getPose().plus(new Transform2d(Units.Inches.of(0), Units.Inches.of(-14.5), new Rotation2d(0)));
+  }
+
   public Angle getRotationMeasure() {
     return Units.Degrees.of(getRotation().getDegrees());
   }
 
+  public boolean reefActionBackwards(Pose2d closestPoseByRotation) {
+    Distance backReefDistance;
+    Distance frontReefDistance;
+    closestPoseByRotation = getClosestPoseByRotation(constField.getReefPositions(true).get());
+    backReefDistance = Units.Meters
+        .of(getBackPose().getTranslation().getDistance(closestPoseByRotation.getTranslation()));
+    frontReefDistance = Units.Meters
+        .of(getFrontPose().getTranslation().getDistance(closestPoseByRotation.getTranslation()));
+    if (backReefDistance.lt(frontReefDistance)) {
+      return true;
+    } else {
+      return false;
+    }
+  }
```

Comment:
Good logic! Can this be made more generic to be able to be useable to support both net and reef (or whatever we want in the future)?